Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_es.cpp

```diff
@@ -261,6 +274,145 @@ IPCCommandResult CWII_IPC_HLE_Device_es::IOCtlV(u32 _CommandAddress)
 
   switch (Buffer.Parameter)
   {
+  case IOCTL_ES_ADDTICKET:
+  {
+    _dbg_assert_msg_(WII_IPC_ES, Buffer.NumberInBuffer == 3,
+                     "IOCTL_ES_ADDTICKET wrong number of inputs");
+
+    INFO_LOG(WII_IPC_ES, "IOCTL_ES_ADDTICKET");
+    std::vector<u8> ticket(Buffer.InBuffer[0].m_Size);
+    Memory::CopyFromEmu(ticket.data(), Buffer.InBuffer[0].m_Address, Buffer.InBuffer[0].m_Size);
+    DiscIO::AddTicket(ticket);
+    break;
+  }
+
+  case IOCTL_ES_ADDTITLESTART:
+  {
+    _dbg_assert_msg_(WII_IPC_ES, Buffer.NumberInBuffer == 4,
+                     "IOCTL_ES_ADDTITLESTART wrong number of inputs");
+
+    INFO_LOG(WII_IPC_ES, "IOCTL_ES_ADDTITLESTART");
+    std::vector<u8> tmd(Buffer.InBuffer[0].m_Size);
+    Memory::CopyFromEmu(tmd.data(), Buffer.InBuffer[0].m_Address, Buffer.InBuffer[0].m_Size);
+
+    constexpr u32 TITLE_ID_TMD_OFFSET = 0x18C;
+    m_addtitle_title_id = Common::swap64(tmd.data() + TITLE_ID_TMD_OFFSET);
+
+    // Write the TMD to title storage.
+    std::string tmd_path = Common::GetTMDFileName(m_addtitle_title_id, Common::FROM_SESSION_ROOT);
+    File::CreateFullPath(tmd_path);
+
+    File::IOFile fp(tmd_path, "wb");
+    fp.WriteBytes(tmd.data(), tmd.size());
```

Comment:
In practice no. But in theory we should be doing more things with the TMD (see later in the file where I explain we don't implement content ID → content index mapping, which would require going through the TMD). So I kept the temporary, it really doesn't matter since this is super-not-performance-critical code in the first place.